dijet/tasks/jersf.py

```diff
@@ -0,0 +1,294 @@
+    # RemoteWorkflow,
+):
+    sandbox = dev_sandbox(law.config.get("analysis", "default_columnar_sandbox"))
+
+    # upstream requirements
+    reqs = Requirements(
+        # RemoteWorkflow.reqs,
+        MergeHistograms=MergeHistograms,
+    )
+
+    # output_collection_cls = law.SiblingFileCollection
+
+    # def create_branch_map(self):
+    #     return {0: None}
+
+    def store_parts(self):
+        parts = super().store_parts()
+        parts.insert_before("version", "datasets", f"datasets_{self.datasets_repr}")
+        return parts
+
+    def requires(self):
+        return {
+            d: self.reqs.MergeHistograms.req(
+                self,
+                dataset=d,
+                branch=-1,
+                _exclude={"branches"},
+                _prefer_cli={"variables"},
+            )
+            for d in self.datasets
+        }
+
+    def workflow_requires(self):
+        reqs = super().workflow_requires()
+        reqs["merged_hists"] = self.requires_from_branch()
+
+        return reqs
+
+    def load_histogram(self, dataset, variable):
+        histogram = self.input()[dataset]["collection"][0]["hists"].targets[variable].load(formatter="pickle")
+        return histogram
+
+    # def reduce_histogram(self, histogram, processes, categories, shifts):
+    def reduce_histogram(self, histogram, processes, shifts):
+        import hist
+
+        def flatten_nested_list(nested_list):
+            return [item for sublist in nested_list for item in sublist]
+
+        # transform into lists if necessary
+        processes = law.util.make_list(processes)
+        # categories = law.util.make_list(categories)
+        shifts = law.util.make_list(shifts)
+
+        # get all leaf categories
+        # category_insts = list(map(self.config_inst.get_category, categories))
+        # leaf_category_insts = set(flatten_nested_list([
+        #     category_inst.get_leaf_categories() or [category_inst]
+        #     for category_inst in category_insts
+        # ]))
+
+        # get all sub processes
+        process_insts = list(map(self.config_inst.get_process, processes))
+        sub_process_insts = set(flatten_nested_list([
+            [sub for sub, _, _ in proc.walk_processes(include_self=True)]
+            for proc in process_insts
+        ]))
+
+        # get all shift instances
+        shift_insts = [self.config_inst.get_shift(shift) for shift in shifts]
+
+        # work on a copy
+        h = histogram.copy()
+
+        # axis selections
+        h = h[{
+            "process": [
+                hist.loc(p.id)
+                for p in sub_process_insts
+                if p.id in h.axes["process"]
+            ],
+            "shift": [
+                hist.loc(s.id)
+                for s in shift_insts
+                if s.id in h.axes["shift"]
+            ],
+        }]
+
+        # axis reductions
+        h = h[{"process": sum, "shift": sum}]
+
+        return h
+
+
+class alpha(HistogramsBaseTask):
```

Comment:
Classes should generally begin with a capital letter. This task seems to do some fitting as well, so maybe rename it to "AlphaExtrapolation"?